zipline/finance/trading.py

```diff
@@ -204,17 +204,18 @@ def normalize_date(self, test_date):
         return pd.tseries.tools.normalize_date(test_date)
 
     def utc_dt_in_exchange(self, dt):
-        return pd.Timestamp(dt).tz_convert(self.exchange_tz)
+        return pd.Timestamp(dt, tz='UTC').tz_convert(self.exchange_tz)
 
     def exchange_dt_in_utc(self, dt):
         return pd.Timestamp(dt, tz=self.exchange_tz).tz_convert('UTC')
 
     def is_market_hours(self, test_date):
-        if not self.is_trading_day(test_date):
+        day = pd.Timestamp(self.utc_dt_in_exchange(test_date).date(), tz='UTC')
```

Comment:
We're doing an awful lot of conversions here...what is this method doing that's currently undesired?